Main/ClientApp/src/Components/Dialogs/ShareCollectionDialog.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useState, ChangeEvent, useEffect } from "react";
+import SimpleDialog from "./SimpleDialog";
+import { shareCollection } from "../../Actions/Actions";
+import { Button, TextField } from "@material-ui/core";
+import Autocomplete from "@material-ui/lab/Autocomplete";
+import { SharedCollectionData } from "../../Model/SharedCollection";
+import UsersStore from "../../Stores/UsersStore";
+
+type ShareCollectionDialogProps = {
+  open: boolean;
+  toggleDialogOpen: () => void;
+  collectionId: number;
+};
+
+export default function ShareCollectionDialog(
+  props: ShareCollectionDialogProps
+) {
+  const title = "Share collection";
+  const description = `You can share this collection with others for them to see and edit. 
+    After sharing, the collection will be visible in their "Shared with me" section.`;
+
+  const createSharedCollectionData = () => {
+    return {
+      collectionId: props.collectionId,
+      userId: selectedUser!.id,
+      viewRights: true,
+      editRights: true,
+    } as SharedCollectionData;
+  };
+
+  const [users, setUsers] = useState(UsersStore.getUsers());
+  useEffect(() => {
+    const changeHandler = () => {
+      setUsers(UsersStore.getUsers());
```

Comment:
I don't think we should display here all users except current. GetRelatedUsers would be more suitable imo

Reply:
Actually it is ok, my bad